### PR TITLE
Fix TS3.9 errors and bugs

### PR DESCRIPTION
**Pull request by felixfbecker**
Detected by TS 3.9

### CODE REVIEW

File: shared/src/api/client/connection.ts [hunk 1]
```diff
@@ -79,7 +79,7 @@ export async function createExtensionHostClientConnection(
         merge(
             of(
                 [...services.model.models.entries()].map(
-                    ([uri, model]): TextModelUpdate => ({ type: 'added', uri, ...model })
+                    ([, model]): TextModelUpdate => ({ type: 'added', ...model })
```
Comment (eseliger) [inline shared/src/api/client/connection.ts:82]: is it always the case that models is a map from model.uri to model?

File: web/src/components/SearchResultMatch.tsx [hunk 2]
```diff
@@ -167,11 +167,11 @@ export class SearchResultMatch extends React.Component<SearchResultMatchProps, S
                 <>
                     {this.state.HTML !== undefined ? (
                         <Link key={this.props.item.url} to={this.props.item.url} className="search-result-match">
-                            {this.bodyIsCode ? (
+                            {this.bodyIsCode() ? (
```
Comment (eseliger) [inline web/src/components/SearchResultMatch.tsx:170]: wow a legit bug?

File: shared/src/api/client/connection.ts (hunk 1 above)
Comment (felixfbecker) [reply shared/src/api/client/connection.ts:82]: Yep.

> A map of all known models, indexed by URI.

The URI of a model is immutable.

File: web/src/components/SearchResultMatch.tsx (hunk 2 above)
Comment (felixfbecker) [reply web/src/components/SearchResultMatch.tsx:170]: Yup! I was amazed too

Comment (felixfbecker) [reply web/src/components/SearchResultMatch.tsx:170]: cc @attfarhan

**Pull request merged**